contracts/settling_game/modules/relics/Relics.cairo

```diff
@@ -97,31 +110,118 @@ func set_relic_holder{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_chec
         tempvar pedersen_ptr = pedersen_ptr;
     }
 
-    // Capture back if held by loser
-    // Check if loser has winners token
-    let (winners_relic_owner) = get_current_relic_holder(winner_token_id);
+    // TODO: Add Order capture back
+
+    let (realms_address) = IModuleController.get_external_contract_address(
+        controller, ExternalContractIds.Realms
+    );
 
-    let (holds_relic) = uint256_eq(winners_relic_owner, loser_token_id);
+    let (winner_realm_data: RealmData) = IRealms.fetch_realm_data(realms_address, winner_token_id);
 
-    if (holds_relic == TRUE) {
-        _set_relic_holder(winner_token_id, winner_token_id);
+    let (loser_relics_len) = owned_relics_count.read(loser_token_id);
+
+    loop_claim_order_relic(0, realms_address, winner_realm_data, loser_token_id, loser_relics_len);
+
+    return ();
+}
+
+func loop_claim_order_relic{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_check_ptr}(
+    index: felt,
+    realms_address: felt,
+    winner_realm_data: RealmData,
+    loser_token_id: Uint256,
+    loser_relics_len: felt,
+) {
+    if (index == loser_relics_len) {
         return ();
     }
 
-    // TODO: Add Order capture back
+    let (relic_id) = owned_relics.read(loser_token_id, index);
+
+    let (relic_owner_data: RealmData) = IRealms.fetch_realm_data(realms_address, relic_id);
+
+    if (winner_realm_data.order == relic_owner_data.order) {
+        _set_relic_holder(relic_id, relic_id);
+        return ();
```

Comment:
Why the early return?

Reply:
Good spot, needs to be removed so loops through all relics. Have also added to test to verify it works for returning multiple relics to same order.